django/db/migrations/optimizer.py

```diff
@@ -49,9 +49,10 @@ def optimize_inner(self, operations, app_label=None):
                 in_between = operations[i + 1:i + j + 1]
                 result = operation.reduce(other, in_between, app_label)
                 if isinstance(result, list):
-                    # Optimize! Add result, then remaining others, then return
-                    new_operations.extend(result)
+                    # Add operations optimized through, optimized operations
```

Comment:
I had to read the code below before I understood what the comment meant. 
Also, perhaps an explanation for why the new ordering helps, would be useful?

For example:

"The operations that were optimized through are moved to the start, since <something about why greater optimisations are possible>, followed by the optimised operations, then any remaining ones."

Also, I found this commit message hard to understand:
```
Refs #27768 -- Ordered operations optimized through before optimized …

…ones.

Operations can only be optimized through if they don't reference any of the
state the operation they are compared against defines or alters.
```

How about:
```
Refs #27768 -- Reversed order of optimized and in-between operations.

Operations can only be optimized through if they don't reference any of the
state the operation they are compared against defines or alters, so it is
safe to reverse the order. 

Doing so allows for greater optimisation due to <X>.
```